FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -58,19 +59,33 @@ public static void ValidateConfiguration(this IServiceCollection services)
                 throw new ConfigurationErrorException("Failed to parse job configuration", ex);
             }
 
-            if (string.IsNullOrEmpty(jobConfiguration.ContainerName))
+            ValidateAgentName(jobConfiguration.AgentName);
+
+            if (!Enum.IsDefined(typeof(QueueType), jobConfiguration.QueueType))
```

Comment:
If the `ConfigurationErrorException` is not necessary here, we can use `EnsureArg.EnumIsDefined` to simplify the check work.

Reply:
Fixed.